admiral/pkg/controller/common/common.go

```diff
@@ -75,7 +76,11 @@ func GetCname(deployment *k8sAppsV1.Deployment, identifier string, nameSuffix st
 		log.Errorf("Unable to get cname for deployment with name %v in namespace %v as it doesn't have the %v annotation", deployment.Name, deployment.Namespace, identifier)
 		return ""
 	}
-	return environment + Sep + alias + Sep + nameSuffix
+	if deployment.Spec.Template.Annotations[AdmiralCnameCaseSensitive] == "true" {
+		log.Infof("admiral.io/cname-case-sensitive annotation enabled on rollout with name %v",deployment.Name)
+		return environment + Sep + alias + Sep + nameSuffix
```

Comment:
Minor thing: Assign `environment + Sep + alias + Sep + nameSuffix` to a variable and use it. I see that line repeated.